internal/tasks/sync_binary.go

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveAdm
+ * Created Date: 2021-11-25
+ * Author: Hailang Mo (wavemomo)
+ */
+
+package tasks
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"github.com/opencurve/curveadm/cli/cli"
+	"github.com/opencurve/curveadm/internal/configure"
+	"github.com/opencurve/curveadm/internal/tasks/task"
+)
+
+type step2CheckoutBinaryVersion struct {
+	containerID           string
+	remoteContainerBinary string
+	localBinary           string
+}
+
+func (s *step2CheckoutBinaryVersion) Execute(ctx *task.Context) error {
+	remoteHostBinaryPath := fmt.Sprintf("/tmp/%s", path.Base(s.localBinary))
+	if err := ctx.Module().Scp(s.localBinary, remoteHostBinaryPath); err != nil {
+		return err
+	}
+
+	_, err := ctx.Module().SshShell("sudo docker cp %s %s:%s", remoteHostBinaryPath, s.containerID,
```

Comment:
Wouldn't it be a problem for users here if they have docker privileges but not sudo privileges?

Reply:
> 

yep, as you said, it will bring hidden dangers, we should provide a configuration item to control sudo privileges.

Reply:
I don't think this issue should be a problem that prevents this pr from merging in, as there are similar issues in the code now.
I've opened a new issue to track this.
What do you think?

Reply:
> 
agree!

Reply:
here #8

Reply:
nice~